.github/workflows/NetCoreTests.yml

```diff
@@ -0,0 +1,57 @@
+name: .NET Core
+
+on: [push, pull_request]
+
+jobs:
+  db:
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - DB: SqlServer2008
+            CONNECTION_STRING: "Server=localhost;initial catalog=nhibernate;User Id=sa;Password=P@ssw0rd;packet size=4096;"
+          - DB: PostgreSQL
+            CONNECTION_STRING: "Host=localhost;Username=nhibernate;Password=nhibernate;Database=nhibernate;Enlist=true;"
+          - DB: Firebird
+            CONNECTION_STRING: "DataSource=localhost;Database=nhibernate;User=SYSDBA;Password=nhibernate;charset=utf8;"
+          - DB: MySQL
+            CONNECTION_STRING: "Server=localhost;Uid=root;Password=nhibernate;Database=nhibernate;Old Guids=True;"
+            ALLOW_FAILURE: true
+          - DB: SQLite
+    runs-on: ubuntu-latest
+    continue-on-error: ${{matrix.ALLOW_FAILURE == true}}
+    env:
+      LANG: en-US.UTF-8 #default POSIX locale doesn't support ignore case comparisons
+    name: ${{matrix.DB}}
+
+    steps:
+    - name: Set up SqlServer
+      if: matrix.DB == 'SqlServer2008'
+      run: |
+        docker run -e "ACCEPT_EULA=Y" -e "SA_PASSWORD=P@ssw0rd" -e "MSSQL_PID=Express" -p 1433:1433 -d --name sqlexpress microsoft/mssql-server-linux:latest;
+
+    - name: Set up MySQL
+      if: matrix.DB == 'MySQL'
+      run: |
+        sudo service mysql stop
+        docker run --name mysql -e MYSQL_ROOT_PASSWORD=nhibernate -e MYSQL_USER=nhibernate -e MYSQL_PASSWORD=nhibernate -e MYSQL_DATABASE=nhibernate -p 3306:3306 --health-cmd="mysqladmin ping" --health-interval=10s --health-timeout=5s --health-retries=3 -d mysql:5.7 mysqld --lower_case_table_names=1
+
+    - name: Set up PostgreSQL
+      if: matrix.DB == 'PostgreSQL'
+      run: |
+       docker run -d -e POSTGRES_USER=nhibernate  -e POSTGRES_PASSWORD=nhibernate -e POSTGRES_DB=nhibernate -p 5432:5432 postgres
+
+    - name: Set up Firebird
+      if: matrix.DB == 'Firebird'
+      run: |
+        docker run --name firebird -e EnableWireCrypt=true -e FIREBIRD_USER=nhibernate -e FIREBIRD_PASSWORD=nhibernate -e ISC_PASSWORD=nhibernate -e FIREBIRD_DATABASE=nhibernate -p 3050:3050 -d jacobalberty/firebird
```

Comment:
Are we sure we want to test against some random container?

Reply:
Not random. Latest stable... But you are right maybe better to  specify 3.0 explicitly - I suspect it would fail on 4.0 for some tests...

Reply:
Or did you mean something else? Like should we trust/depend on some not official container? It's open source container  - I see no issue with it. It works for us - so why not...

Reply:
I also saw Oracle container: https://hub.docker.com/r/gvenzl/oracle-xe. I think we should add it too sometime later.

Reply:
>  Like should we trust/depend on some not official container?

Yes, I meant that.

> It's open source container with 1M+ downloads

It does not mean anything, really.